ngraph_bridge/ngraph_builder.cc

```diff
@@ -2547,48 +2547,93 @@ static Status TranslatePackOp(const Node* op, const std::vector<const Tensor*>&,
   return Status::OK();
 }
 
-static Status TranslatePadOp(const Node* op,
+// Helper function for several Pad Ops
+static Status _CreateNgPadOp(const Node* op,
                              const std::vector<const Tensor*>& static_input_map,
                              Builder::OpMap& ng_op_map) {
-  shared_ptr<ng::Node> ng_input, ng_paddings_op;
-  TF_RETURN_IF_ERROR(GetInputNodes(ng_op_map, op, &ng_input, &ng_paddings_op));
+  shared_ptr<ng::Node> ng_input, ng_paddings_op, pad_val_op, result_pad_op;
 
-  std::vector<int64> paddings;
-  TF_RETURN_IF_ERROR(GetStaticInputVector(op, 1, static_input_map, &paddings));
+  // Set inputs and pad_val_op
+  if (op->type_string() == "Pad" || op->type_string() == "MirrorPad") {
```

Comment:
Nice. Now that you've a common helper for all 3 cases, you could just rename it to `TranslatePadOp` and use it directly in `TRANSLATE_OP_MAP` for all 3 pad ops?

Reply:
Oops, i missed that. Sure.